homework_14_140123/src/MethodsImplementation.java

```diff
@@ -0,0 +1,409 @@
+            System.out.println(file + " is created");
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+        BufferedWriter bufferedWriter = null;
+        try {
+            bufferedWriter = new BufferedWriter(new FileWriter(file));
+            bufferedWriter.write("First line\n");
+            bufferedWriter.write("Second line\n");
+            bufferedWriter.write("Third line\n");
+        } catch (IOException e) {
+            System.out.println(e);
+        } finally {
+            try {
+                bufferedWriter.close();
+            } catch (IOException e) {
+                System.out.println(e);
+            }
+        }
+        file.renameTo(new File("NewNameFile.txt"));
+    }
+
+
+    //Task6
+//    Создать текстовый файл, но с условием, что при достижении 12ого символа запись будет производиться с новой строки,
+//    потом считать содержимое файла и вывести на екран.
+//    Пользователь вводит текст из консоли
+    public static void newLineTransition() {
+        File file = new File("TextFileTask6.txt");
+        BufferedWriter bufferedWriter = null;
+        try {
+            file.createNewFile();
+            System.out.println(file + " is created");
+            bufferedWriter = new BufferedWriter(new FileWriter(file, true));
+            System.out.println("Enter the text:");
+            Scanner scanner = new Scanner(System.in);
+            String line = scanner.nextLine();
+            char[] arrChar = line.toCharArray();
+            String[] arrString = new String[arrChar.length];
+            for (int i = 0; i < arrString.length; i++) {
+                arrString[i] = String.valueOf(arrChar[i]);
+            }
+            for (int i = 0; i < arrString.length; i = i + 12) {
+                if (i == 0) {
+                    i = i + 11;
+                    arrString[i] = arrString[i] + "\n";
+                } else {
+                    arrString[i] = arrString[i] + "\n";
+                }
+            }
+            String line2 = String.join("", arrString);
+            bufferedWriter.write(line2);
+        } catch (IOException e) {
+            System.out.println(e);
+        } finally {
+            try {
+                bufferedWriter.close();
+            } catch (IOException e) {
+                System.out.println(e);
+            }
+        }
+        BufferedReader bufferedReader = null;
+        try {
+            bufferedReader = new BufferedReader(new FileReader(file));
+            StringBuilder stringBuilder = new StringBuilder();
+            while (true) {
+                int i;
+                if ((i = bufferedReader.read()) == -1) {
+                    break;
+                } else {
+                    stringBuilder.append((char) i);
+                }
+            }
+            System.out.println(stringBuilder);
+        } catch (IOException e) {
+            System.out.println(e);
+        } finally {
+            try {
```

Comment:
you don't need try block here, just close stream

Reply:
fixed. using try with resources.